rs/ic-observability/multiservice-discovery/README.md

```diff
@@ -12,6 +12,31 @@ networks on the fly.
 Containers built by the DRE repository CI are published to [GHCR](https://ghcr.io/dfinity/dre/).
 They are only built from PRs stemming from branches named `container-*`.
 
+## Test
+
+We are performing these checks to ensure that the multiservice-discovery fetches all the necessary targets
+before deploying it in production.
+If less or erroneus targets end up in production, we risk compromising the observability stack leading
+to incomplete or erroneus observability data.
+If the mainnet registry gets updated or if the logic for target discovery is modified,
+this assertions should be updated accordingly.
+One can run the multiservice-discovery from local and manually update the targets after inspection:
+
+```bash
+mkdir /tmp/mainnet-registry
+cargo run --package multiservice-discovery -- --targets-dir /tmp/mainnet-registry
+curl -L -v http://localhost:8000/prom/targets > targets.json
+
+# Check number of targets:
+jq -r '[.[] | .targets[]] | length' targets.json
+
+# List unique jobs:
+jq -r '.[].labels.job' targets.json | sort -u
+
+# List unique subnets:
+jq -r '.[].labels.ic_subnet'  targets.json | sort -u
+```
```

Comment:
Is there a particular reason why we use prom format here? Feels like our format for displaying targets is more natural and a must-have feature while this is just a convenience that we added.

Reply:
Probably not needed anymore when we switch to public dashboard API